tests/test_optional_enhance.py

```diff
@@ -0,0 +1,119 @@
+import pytest
+from swap_meet.vendor import Vendor
+from swap_meet.item import Item
+from swap_meet.clothing import Clothing
+from swap_meet.decor import Decor
+from swap_meet.electronics import Electronics
+
+
+def test_newest():
+    item_a = Clothing(age=1.0)
+    item_b = Decor(age=2.0)
+    item_c = Clothing(age=4.0)
+    item_d = Decor(age=5.0)
+    item_e = Clothing(age=3.0)
+    tai = Vendor(
+        inventory=[item_a, item_b, item_c, item_d, item_e]
+    )
```

Comment:
The `Generic` approach really doesn't like this `__init__` call. It doesn't match the `Item`-derived types properly.

By completely separating the `str` and `Item` versions, this "just works". If you want to look into getting the `Generic` version to work here, try looking into the concepts of covariance and contravariance. I'm not sure if the type hinting in Python is robust enough to handle this, though.